packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -424,14 +564,89 @@ abstract class Layer extends AbstractNode with DiagnosticableTreeMixin {
+/// disposed.
+///
+/// [Layer] objects retain native resourses such as [EngineLayer]s and [Picture]
+/// objects. These objects may in turn retain large chunks of texture memory,
+/// either directly or indirectly.
+///
+/// The layer's native resources must be retained as long as there is some
+/// object that can add it to a scene. Typically, this is either its
+/// [Layer.parent] or an undisposed [RenderObject] that will append it to a
+/// [ContainerLayer]. Layers automatically hold a handle to their children, and
+/// RenderObjects automatically hold a handle to their [RenderObject.layer] as
+/// well as any [PictureLayer]s that they paint into using the
+/// [PaintingContext.canvas]. A layer automatically releases its resources once
+/// at least one handle has been acquired and all handles have been disposed.
+/// [RenderObject]s that create additional layer objects must manually manage
+/// the handles for that layer similarly to the implementation of
+/// [RenderObject.layer].
+///
+/// [RenderObject]s automatically store handles to [PictureLayer](s) they draw
+/// into and dispose of those handles when the [RenderObject.dispose] method
+/// is called or they otherwise no longer need the reference to that
+/// [PictureLayer] (such as when they paint a new picture).
+///
+/// A handle is also automatically managed for [RenderObject.layer].
+///
+/// If a [RenderObject] creates layers in addition to its [RenderObject.layer],
+/// it must create a handle to that layer and dispose of it when the layer is
+/// no longer needed. For example, if it re-creates or nulls out an existing
+/// layer in [RenderObject.paint], it should dispose of the handle to the
+/// old layer. It should also dispose of any layer handles it holds in
+/// [RenderObject.dispose].
+class LayerHandle<T extends Layer?> {
+  /// Creates a new LayerHandle, optionally referencing a [Layer].
+  LayerHandle([this._layer]) {
+    if (_layer != null) {
+      _layer!._refCount += 1;
+    }
+  }
+
+  T? _layer;
+
+  /// The [Layer] this class holds.
+  ///
+  /// Setting a new value will dispose the previously held layer if there are
+  /// no other open handles to that layer.
+  ///
+  /// To dispose of this handle, set this value to `null`.
+  T? get layer => _layer;
+
+  set layer(T? layer) {
+    assert(
+      layer?.debugDisposed != true,
+      'Attempted to create a handle to an already disposed layer: $layer.',
+    );
+    if (identical(layer, _layer)) {
+      return;
+    }
+    _layer?._unref();
+    _layer = layer;
+    if (_layer != null) {
+      _layer!._refCount += 1;
+    }
   }
+
+  @override
+  String toString() => 'LayerHandle(${_layer != null ? _layer.toString() : 'DISPOSED'})';
 }
 
 /// A composited layer containing a [Picture].
 ///
-/// Picture layers are always leaves in the layer tree.
+/// Picture layers are always leaves in the layer tree. They are also
+/// responsible for disposing of the [Picture] object they hold. This is done
```

Comment:
nit: "This is _typically_ done" (in case some users don't use the upper layers of the framework, e.g. perhaps they have their own custom render object system)